yarn-project/circuits.js/src/structs/__snapshots__/root_rollup.test.ts.snap

```diff
@@ -1,142 +1,134 @@
+end_contract_tree_snapshot:
+root: 0x900
+next_available_leaf_index: 900
 
-proof: [ 50 50 50 50 50 50 50 50 50 50 50 50 50 50 50 50 ]
-vk: key.composer_type: 0
-key.circuit_size: 65
-key.num_public_inputs: 66
-key.commitments: [
- A: { 0x200, 0x300 }
-]
-key.contains_recursive_proof: 1
-key.recursive_proof_public_input_indices: [ 190 191 192 193 194 ]
+calldata_hash: [ 0x1001 0x1002 ]
 
+proof: [ 50 50 50 50 50 50 50 50 50 50 50 50 50 50 50 50 ]
+vk: 0x12b300
 vk_index: 110
 vk_sibling_path: leaf_index: 120
 sibling_path: [ 0x120 0x121 0x122 0x123 0x124 0x125 0x126 0x127 ]
 
- merge_rollup_public_inputs: end_aggregation_object:
-P0: { 0x1100, 0x1101 }
-P1: { 0x1200, 0x1201 }
+ base_or_merge_rollup_public_inputs: rollup_type:
+0
+rollup_subtree_height:
+0x1100
+end_aggregation_object:
+P0: { 0x1200, 0x1201 }
+P1: { 0x1300, 0x1301 }
 public_inputs: [
- 0x1102
- 0x1103
- 0x1104
- 0x1105
+ 0x1202
+ 0x1203
+ 0x1204
+ 0x1205
 ]
-proof_witness_indices: [ 1106 1107 1108 1109 110a 110b ]
+proof_witness_indices: [ 1206 1207 1208 1209 120a 120b ]
 has_data: 0
 
 constants:
 start_tree_of_historic_private_data_tree_roots_snapshot:
- root: 0x1200
-next_available_leaf_index: 1200
-
-start_tree_of_historic_contract_tree_roots_snapshot:
-root: 0x1300
+ root: 0x1300
 next_available_leaf_index: 1300
 
-tree_of_historic_l1_to_l2_msg_tree_roots_snapshot:
+start_tree_of_historic_contract_tree_roots_snapshot:
 root: 0x1400
 next_available_leaf_index: 1400
 
-private_kernel_vk_tree_root: 0x1501
-public_kernel_vk_tree_root: 0x1502
-base_rollup_vk_hash: 0x1503
-merge_rollup_vk_hash: 0x1504
+tree_of_historic_l1_to_l2_msg_tree_roots_snapshot:
+root: 0x1500
+next_available_leaf_index: 1500
+
+private_kernel_vk_tree_root: 0x1601
+public_kernel_vk_tree_root: 0x1602
+base_rollup_vk_hash: 0x1603
+merge_rollup_vk_hash: 0x1604
 
 start_private_data_tree_snapshot:
-root: 0x1300
-next_available_leaf_index: 1300
+root: 0x1400
+next_available_leaf_index: 1400
 
 end_private_data_tree_snapshot:
-root: 0x1300
-next_available_leaf_index: 1300
+root: 0x1400
+next_available_leaf_index: 1400
 
 start_nullifier_tree_snapshot:
-root: 0x1500
-next_available_leaf_index: 1500
-
-end_nullifier_tree_snapshots:
 root: 0x1600
 next_available_leaf_index: 1600
 
-start_contract_tree_snapshot:
+end_nullifier_tree_snapshots:
 root: 0x1700
 next_available_leaf_index: 1700
 
-end_contract_tree_snapshot:
+start_contract_tree_snapshot:
 root: 0x1800
 next_available_leaf_index: 1800
 
-calldata_hash: [ 0x1901 0x1902 ]
+end_contract_tree_snapshot:
+root: 0x1900
+next_available_leaf_index: 1900
 
-proof: [ 50 50 50 50 50 50 50 50 50 50 50 50 50 50 50 50 ]
-vk: key.composer_type: 0
-key.circuit_size: 65
-key.num_public_inputs: 66
-key.commitments: [
- A: { 0x200, 0x300 }
-]
-key.contains_recursive_proof: 1
-key.recursive_proof_public_input_indices: [ 190 191 192 193 194 ]
+calldata_hash: [ 0x2001 0x2002 ]
 
+proof: [ 50 50 50 50 50 50 50 50 50 50 50 50 50 50 50 50 ]
+vk: 0x133e40
```

Comment:
This seems like you're serializing the pointer to the VK instead of the VK itself. Maybe there is a pointer dereference missing when cout'ing the VK from the struct?